File: 1200-minimum-absolute-difference/1200-minimum-absolute-difference.java
class Solution {
    public List<List<Integer>> minimumAbsDifference(int[] arr) {
        List<List<Integer>> res = new ArrayList();
        
        Arrays.sort(arr);
        
        int mini = Integer.MAX_VALUE;
        for(int i=1; i<arr.length; i++)
            mini = Math.min(arr[i]-arr[i-1], mini);
        
        for(int i=1; i<arr.length; i++){
            if(arr[i]-arr[i-1] == mini){
                List<Integer> vec = new ArrayList();
                vec.add(arr[i-1]);
                vec.add(arr[i]);
                res.add(vec);
            }
        }
        return res;
    }
}

File: 1200-minimum-absolute-difference/1200-minimum-absolute-difference.cpp
class Solution {
public:
    vector<vector<int>> minimumAbsDifference(vector<int>& arr) {
        vector<vector<int>> res;
        
        sort(arr.begin(), arr.end());
        
        int mini = INT_MAX;
        for(int i=1; i<arr.size(); i++)
            mini = min(arr[i]-arr[i-1], mini);
        
        for(int i=1; i<arr.size(); i++){
            if(arr[i]-arr[i-1] == mini){
                vector<int> vec;
                vec.push_back(arr[i-1]);
                vec.push_back(arr[i]);
                res.push_back(vec);
            }
        }
        return res;
    }
};